storage/range.go

```diff
@@ -838,46 +864,46 @@ func (r *Range) applyRaftCommand(index uint64, originNodeID multiraft.NodeID, ar
 	return reply.Header().GoError()
 }
 
-// startGossip periodically gossips the cluster ID if it's the first
-// range. This function should only be called if this range replica is
-// the raft leader.
-func (r *Range) startGossip(stopper *util.Stopper) {
-	stopper.RunWorker(func() {
-		r.maybeGossipClusterID()
-		r.maybeGossipFirstRange()
-
-		ticker := time.NewTicker(ttlClusterIDGossip / 2)
-		for {
-			select {
-			case <-ticker.C:
-				if held, expired := r.HasLeaderLease(r.rm.Clock().Now()); held && !expired {
-					r.maybeGossipClusterID()
-					r.maybeGossipFirstRange()
-				}
-			case <-stopper.ShouldStop():
-				return
-			}
-		}
-	})
-}
+// maybeGossipFirstRangeWithLease is periodically called and gossips the
+// sentinel and first range metadata if the range has the lease and is the
+// first range. Since one replica in the first range should always gossip this
+// information, a lease is acquired if there is no active lease.
+func (r *Range) maybeGossipFirstRangeWithLease() {
+	// If no Gossip available (some tests) or not first range: nothing to do.
+	if r.rm.Gossip() == nil || !r.isInitialized() || !r.IsFirstRange() {
+		return
+	}
+	timestamp := r.rm.Clock().Now()
+	held, shouldGetLease := r.HasLeaderLease(timestamp)
+
+	// Someone has an active lease; they'll do the work, not us.
+	if !held && !shouldGetLease {
+		return
+	}
 
-// maybeGossipClusterID gossips the cluster ID if this range is
-// the start of the key space.
-func (r *Range) maybeGossipClusterID() {
-	if r.rm.Gossip() != nil {
-		if err := r.rm.Gossip().AddInfo(gossip.KeyClusterID, r.rm.ClusterID(), ttlClusterIDGossip); err != nil {
-			log.Errorf("failed to gossip cluster ID %s: %s", r.rm.ClusterID(), err)
+	if shouldGetLease {
+		// We don't have an active lease, but we think we should be
```

Comment:
This comment implies that it would be unexpected for us to fail to acquire the lease, when in fact that is the more likely outcome (all the replicas will hit this at the same time at startup). Change the comment to "No one has an active lease. Try to acquire it, and if we succeed, we can gossip". If we had a separate error type for rejected lease attempts then we could log a warning for any other errors (and downgrade the "could not acquire" message to V(1)).